src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts

```diff
@@ -0,0 +1,480 @@
+
+        popoverUtils.longWithHover($("#replication-filtering-info"),
+            {
+                content:
+                    "<ul class='margin-bottom margin-bottom-xs'>" +
+                        "<li><small>Check this toggle in order to be able to define filtering on this Replication Hub task.</small></li>" +
+                        "<li><small>You will be able to define the replication filtering when editing the task after saving this configuration.</small></li>" +
+                        "<li><small class='text-warning'><i class='icon-warning'></i>" +
+                            "<span>Note: This parameter cannot be modified after saving this configuration.</span></small>" +
+                        "</li>" +
+                    "</ul>"
+            });
+    }
+    
+    private loadPossibleMentors() {
+        return new getPossibleMentorsCommand(this.activeDatabase().name)
+            .execute()
+            .done(mentors => this.possibleMentors(mentors));
+    }
+
+    private initObservables() {
+        const model = this.editedHubTask();
+        
+        this.dirtyFlag = new ko.DirtyFlag([
+            model.taskName,
+            model.manualChooseMentor,
+            model.mentorNode,
+            model.delayReplicationTime,
+            model.showDelayReplication,
+            model.withFiltering,
+            model.allowReplicationFromHubToSink,
+            model.allowReplicationFromSinkToHub
+        ], false, jsonUtil.newLineNormalizingHashFunction)
+    }
+
+    compositionComplete() {
+        super.compositionComplete();
+        document.getElementById('taskName').focus();
+        
+        $('.edit-pull-replication-hub-task [data-toggle="tooltip"]').tooltip(); 
+    }
+
+    saveReplicationHubTask() {
+        if (!this.isValid(this.editedHubTask().validationGroupForSave)) {
+            return;
+        }
+
+        this.spinners.saveHubTask(true);
+
+        const dto = this.editedHubTask().toDto(this.taskId);
+        this.taskId = this.isNewTask() ? 0 : this.taskId;
+
+        eventsCollector.default.reportEvent("pull-replication-hub", "save");
+
+        new saveReplicationHubTaskCommand(this.activeDatabase(), dto)
+            .execute()
+            .done((result: Raven.Client.Documents.Operations.OngoingTasks.ModifyOngoingTaskResult) => {
+                this.dirtyFlag().reset();
+               
+                if (this.isNewTask() && this.canDefineCertificates) {
+                    // don't navigate back to list, allow user to add replication access
+                    this.isNewTask(false);
+                    this.taskId = result.TaskId;
+                    this.addNewReplicationAccess();
+                } else {
+                    this.goToOngoingTasksView();
+                }
+                
+            })
+            .always(() => this.spinners.saveHubTask(false));
+    }
+
+    saveReplicationAccessItem() {
+        if (this.editedHubTask().withFiltering()) {
```

Comment:
do we have info about filtered/not filtered replication in editedReplicationAccessItem ? If yes then we can create observable which will choose proper validation group and here we don't care about editedReplicationAccessItem internals.

Reply:
ok - implemented in another simple way - w/o duplicating data